src/libproc_macro/lib.rs

```diff
@@ -459,14 +423,39 @@ pub enum Spacing {
 }
 
 /// A literal character (`'a'`), string (`"hello"`), or number (`2.3`).
-#[derive(Clone, Debug)]
+#[derive(Clone)]
 #[unstable(feature = "proc_macro", issue = "38356")]
-pub struct Literal(token::Token);
+pub struct Literal {
+    #[unstable(feature = "proc_macro_internals", issue = "27812")]
+    #[doc(hidden)]
+    pub kind: LiteralKind,
+
+    #[unstable(feature = "proc_macro_internals", issue = "27812")]
+    #[doc(hidden)]
+    pub contents: Term,
```

Comment:
Do we want to commit to using interning for literals? I'm not sure interning it terrible useful for that.

Reply:
Wait, I can make these fields private again. Definitely not considering stabilizing any of this.